cmd/lncli/commands.go

```diff
@@ -711,15 +881,45 @@ var AddInvoiceCommand = cli.Command{
 }
 
 func addInvoice(ctx *cli.Context) error {
+	var (
+		preimage []byte
+		receipt  []byte
+		value    int64
+		err      error
+	)
+
 	client, cleanUp := getClient(ctx)
 	defer cleanUp()
 
-	preimage, err := hex.DecodeString(ctx.String("preimage"))
+	args := ctx.Args()
+
+	switch {
+	case ctx.IsSet("value"):
+		value = ctx.Int64("value")
+	case args.Present():
+		value, err = strconv.ParseInt(args.First(), 10, 64)
+		args = args.Tail()
+		if err != nil {
+			return fmt.Errorf("unable to decode value argument: %v", err)
+		}
+	default:
+		return fmt.Errorf("value argument missing")
+	}
+
+	switch {
+	case ctx.IsSet("preimage"):
+		preimage, err = hex.DecodeString(ctx.String("preimage"))
+	case args.Present():
+		preimage, err = hex.DecodeString(args.First())
+	default:
+		return fmt.Errorf("preimage argument missing")
```

Comment:
The `preimage` value isn't actually mandatory. If left blank, it'll be generated by the `lnd` node the call is being made to. As is, this PR breaks this existing functionality. Thankfully, it can be remedied simply by removing this `default` case.